enhancements/installer/openstack-supported-versions.md

```diff
@@ -0,0 +1,217 @@
+
+- [OpenStack supported versions](#openstack-supported-versions)
+  - [Release Signoff Checklist](#release-signoff-checklist)
+  - [Summary](#summary)
+  - [Motivation](#motivation)
+    - [Goals](#goals)
+    - [Non-Goals](#non-goals)
+  - [Proposal](#proposal)
+    - [Design Details](#design-details)
+    - [User Stories](#user-stories)
+      - [Check the OpenStack version](#check-the-openstack-version)
+      - [Ignore the version check](#ignore-the-version-check)
+        - [Example Usage](#example-usage)
+    - [Test Plan](#test-plan)
+    - [Graduation Criteria](#graduation-criteria)
+      - [Dev Preview -> Tech Preview](#dev-preview---tech-preview)
+      - [Tech Preview -> GA](#tech-preview---ga)
+    - [Infrastructure Needs](#infrastructure-needs)
+    - [Drawbacks](#drawbacks)
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [x] Test plan is defined
+- [x] Graduation criteria
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Summary
+
+OpenShift can only be installed on supported versions of OpenStack.
+While the [Support Matrix](https://access.redhat.com/articles/4679401) is
+well documented, the OpenShift installer doesn't check which version of
+OpenStack is running before attempting an operation (e.g. installation). 
+Running OpenShift over an unsupported version of OpenStack is problematic
+and can lead to unstable infrastructure. To avoid that, it is proposed
+to programmatically verify the version of OpenStack is supported
+otherwise stop and fail.
+
+## Motivation
+
+### Goals
+
+- Disallow any kind of OpenShift operation done through the installer to run
+  against a cloud where the OpenStack version is not supported.
+- Allow an administrator to ignore the OpenStack version check and force the
+  installer to proceed anyway. This can be useful if the OpenStack cloud
+  is partially upgraded to a major version or if there is a Support
+  Exception granted to run that deprecated version of OpenStack.
+
+### Non-Goals
+
+This enhancement does not intend do block any OpenShift operations done
+outside of the installer if the OpenStack version is not supported.
+
+## Proposal
+
+In order to implement this feature fully, the following changes must be made
+in the openshift/installer:
+
+- Update gophercloud vendoring to (at least) [b56bf1](https://github.com/gophercloud/gophercloud/pull/2037/commits/5a1daf082451587459058cc539c53f2ae1aaaef3)
+  so we can benefit from the apiversions package.
+- Create a constant that will list all the supported versions
+  of Compute API based on the upstream manual.
+- Add a new validation against the OpenStack cloud to make sure that
+  the Compute API version is supported.
+- Create a configuration flag `ignoreVersionCheck` that when set to
+  false will send a warning if the OpenStack version isn't supported and
+  proceed anyway.
+
+### Design Details
```

Comment:
This section currently details _how_ to check.

It's missing details on _when_ the check is performed, and more precisely: what the trigger is, and what logic parts must be changed. IIUC, the "install" check should live in the Installer pre-flight checks, whereas the "upgrade" check should live in the `oc adm` binary. Does that make sense?

Reply:
yes, i'll update wording.